### PR TITLE
[docs] EAS Build: update infra docs - macOS Monterey image

### PR DESCRIPTION
**Pull request by dsokal**
**Merge after https://github.com/expo/turtle-v2/pull/779 is merged and deployed.**

# Why

- Follow-up to https://github.com/expo/turtle-v2/pull/779
- macOS Catalina will no longer be supported
- Adding macOS Monterey image.

# How

Update docs.

# Test Plan

None

### CODE REVIEW

File: docs/pages/build-reference/infrastructure.md [hunk 1]
```diff
@@ -15,16 +15,16 @@ When selecting an image for the build you can use the full name provided below o
 - `default` alias will be assigned to the environment that most closely resembles the configuration used for Expo SDK development.
 - `latest` alias will be assigned to the image with the most up to date versions of the software.
 
-> **Note:** If you don't provide `image` in eas.json, your build is going to use the `default` image. There is one exception to this rule - if you have a managed project and you don't specify `image`, it will be chosen based on your Expo SDK version. E.g. SDKs 41 and lower use `macos-catalina-10.15-xcode-12.1`, SDK 42 uses `macos-big-sur-11.4-xcode-12.5`, and SDK 43 uses `macos-big-sur-11.4-xcode-13.0`.
+> **Note:** If you don't provide `image` in eas.json, your build is going to use the `default` image. There is one exception to this rule - if you have a managed project and you don't specify `image`, it will be chosen based on your Expo SDK version. E.g. SDK 42 and lower uses `macos-big-sur-11.4-xcode-12.5`, and SDK 43 uses `macos-big-sur-11.4-xcode-13.0`.
```
Comment (kbrandwijk) [inline docs/pages/build-reference/infrastructure.md:18]: Which one does 44 use?

File: docs/pages/build-reference/infrastructure.md [hunk 2]
```diff
@@ -15,16 +15,16 @@ When selecting an image for the build you can use the full name provided below o
 - `default` alias will be assigned to the environment that most closely resembles the configuration used for Expo SDK development.
 - `latest` alias will be assigned to the image with the most up to date versions of the software.
```
Comment (kbrandwijk) [inline docs/pages/build-reference/infrastructure.md:16]: If Big Sur stays 'latest' then this statement is false.

**Review by kbrandwijk — APPROVED** [review]

**Review by wkozyra95 — APPROVED** [review]

File: docs/pages/build-reference/infrastructure.md (hunk 1 above)
Comment (dsokal) [reply docs/pages/build-reference/infrastructure.md:18]: The default one.

File: docs/pages/build-reference/infrastructure.md [hunk 3]
```diff
@@ -15,16 +15,16 @@ When selecting an image for the build you can use the full name provided below o
...
 
-> **Note:** If you don't provide `image` in eas.json, your build is going to use the `default` image. There is one exception to this rule - if you have a managed project and you don't specify `image`, it will be chosen based on your Expo SDK version. E.g. SDKs 41 and lower use `macos-catalina-10.15-xcode-12.1`, SDK 42 uses `macos-big-sur-11.4-xcode-12.5`, and SDK 43 uses `macos-big-sur-11.4-xcode-13.0`.
+> **Note:** If you don't provide `image` in **eas.json** for a bare workflow project, your build is going to use the `default` image. For managed workflow projects, the image will be chosen automatically unless you provide `image`.
```
Comment (wkozyra95) [inline docs/pages/build-reference/infrastructure.md:18]:
```suggestion
> **Note:** If you don't provide `image` in **eas.json** for a bare workflow project, your build is going to use the `default` image. For managed workflow projects, the default image will be chosen automatically based on the project's SDK version.
```

**Review by wkozyra95 — APPROVED** [review]

**Pull request merged**